ibis/pandas/execution/window.py

```diff
@@ -469,5 +502,11 @@ def execute_series_dense_rank(op, data, **kwargs):
 
 @execute_node.register(ops.PercentRank, (pd.Series, SeriesGroupBy))
 def execute_series_percent_rank(op, data, **kwargs):
+    # TODO(phillipc): Handle ORDER BY
+    return (data.rank(method='min') - 1) / (data.apply(_win_count) - 1)
```

Comment:
I'm very confused about this. `data.apply(_win_count)` is just returning a `DataFrame` of the same size as `data`, but were all the items on it are the length of  `data`, right?
```python
>>> df = pandas.DataFrame({'foo': [1, 2, 3],
...                        'bar': [4, 5, 6]})
>>> df.apply(_win_count)
   foo  bar
0    3    3
1    3    3
2    3    3
```
And then you subtract `1`, and divide the rank dataframe by it. So, it feels that what you're doing is a very complex way of:
```python
return (data.rank(method='min') - 1) / (len(data) - 1)
```
What am I missing?

Reply:
It needs to keep the same index. do you have any suggestions?

```python
>>> df_ex = pd.DataFrame({'a': [10, 11, 12]}, index=[1,2,3])
>>> print(len(df_ex))
>>> print('=' * 10)
>>> print(win_count(df_ex))

3
==========
   a
1  3
2  3
3  3
```

maybe it doesn't look important in this example .. but when it uses `SeriesGroupBy`, each chunk could have different counts.

Reply:
Is `transform` doing what you want?
```python
data = pandas.DataFrame({'name': ['foo', 'bar', 'foo'],
                       'value': [1, 2, 3]})

g = data.groupby('name')['value']
print(g.apply(_win_count))
print(g.transform(len))
```
I guess you'll need an `if` and address separately when `data` is a `DataFrame`, or a group by object, but this should be simpler.

Reply:
thanks for the suggestion @datapythonista ! changes applied.